cpp/benchmarks/text/jaccard.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+
+#include <cudf/strings/strings_column_view.hpp>
+
+#include <nvtext/jaccard.hpp>
+
+#include <nvbench/nvbench.cuh>
+
+#include <rmm/device_buffer.hpp>
+
+static void bench_jaccard(nvbench::state& state)
+{
+  auto const num_rows        = static_cast<cudf::size_type>(state.get_int64("num_rows"));
+  auto const row_width       = static_cast<cudf::size_type>(state.get_int64("row_width"));
+  auto const substring_width = static_cast<cudf::size_type>(state.get_int64("substring_width"));
+
+  if (static_cast<std::size_t>(num_rows) * static_cast<std::size_t>(row_width) >=
+      static_cast<std::size_t>(std::numeric_limits<cudf::size_type>::max())) {
+    state.skip("Skip benchmarks greater than size_type limit");
+  }
+
+  data_profile const strings_profile =
+    data_profile_builder()
+      .distribution(cudf::type_id::STRING, distribution_id::NORMAL, 0, row_width)
+      .no_validity();
```

Comment:
I don't know if it matters for the immediate applications at hand, but do we want to see what happens to perf when there is a long-tailed distribution? I would expect "real" documents to have a small number of extremely large strings, which I'd expect to perform substantially worse (not something we have to address in this PR, I'm asking for benchmarks so that we can be aware of it).

Reply:
https://github.com/rapidsai/cudf/issues/13726